scipy/sparse/linalg/eigen/tests/test_svds.py

```diff
@@ -363,35 +528,59 @@ def test_svd_LM_zeros_matrix_gh_3452(self):
 
 # --- Perform tests with each solver ---
 
+class Test_SVDS_once():
+    @pytest.mark.parametrize("solver", ['ekki', object])
+    def test_svds_input_validation_solver(self, solver):
+        message = "solver must be one of"
+        with pytest.raises(ValueError, match=message):
+            svds(np.ones((3, 4)), k=2, solver=solver)
+
 class Test_SVDS_ARPACK(SVDSCommonTests):
 
     def setup_method(self):
         self.solver = 'arpack'
 
-    def test_svd_maxiter(self):
-        # check that maxiter works as expected
-        x = hilbert(6)
-        # ARPACK shouldn't converge on such an ill-conditioned matrix with just
-        # one iteration
-        assert_raises(ArpackNoConvergence, svds, x, 1, maxiter=1, ncv=3)
-        # but 100 iterations should be more than enough
-        u, s, vt = svds(x, 1, maxiter=100, ncv=3)
-        assert_allclose(s, [1.7], atol=0.5)
 
+    @pytest.mark.parametrize("ncv", list(range(-1, 8)) + [4.5, "5"])
+    def test_svds_input_validation_ncv_1(self, ncv):
+        rng = np.random.default_rng(0)
+        A = rng.random((6, 7))
+        k = 3
+        if ncv in {4, 5}:
+            u, s, vh = svds(A, k=k, ncv=ncv, solver=self.solver)
+        # partial decomposition, so don't check that u@diag(s)@vh=A;
+        # do check that scipy.sparse.linalg.svds ~ scipy.linalg.svd
+            _check_svds(A, k, u, s, vh, check_usvh_A=False, check_svd=True)
+        else:
+            message = ("`ncv` must be an integer satisfying")
+            with pytest.raises(ValueError, match=message):
+                svds(A, k=k, ncv=ncv, solver=self.solver)
+
+    def test_svds_input_validation_ncv_2(self):
+        # I think the stack trace is reasonable when `ncv` can't be converted
+        # to an int.
+        message = "int() argument must be a"
+        with pytest.raises(TypeError, match=re.escape(message)):
+            svds(np.eye(10), ncv=[], solver=self.solver)
+
+        message = "invalid literal for int()"
+        with pytest.raises(ValueError, match=message):
+            svds(np.eye(10), ncv="hi", solver=self.solver)
+
+
+    # I can't see a robust relationship between `ncv` and relevant outputs
+    # (e.g. accuracy, time), so no test of the parameter.
```

Comment:
It's annoying to have a parameter that's only accepted by ARPACK and doesn't seem to do anything.